oak-run/src/main/java/org/apache/jackrabbit/oak/run/DataStoreCheckCommand.java

```diff
@@ -242,16 +248,26 @@ static int checkDataStore(String... args) {
             }
 
             if (options.has(refOp) || options.has(consistencyOp)) {
-                if (options.has(verbose) &&
+
+                // Find blob ids by traversal for verbose mode + Segment Store or if verboseRootPath option
+                // is present (find blob references under a specific root path.)
+                if ((options.has(verbose) &&
                     (nodeStore instanceof SegmentNodeStore ||
-                        nodeStore instanceof org.apache.jackrabbit.oak.segment.SegmentNodeStore)) {
+                        nodeStore instanceof org.apache.jackrabbit.oak.segment.SegmentNodeStore)) || options.has(verboseRootPath)) {
```

Comment:
Does this mean there are two instanceof checks?

    (nodeStore instanceof SegmentNodeStore ||
    nodeStore instanceof org.apache.jackrabbit.oak.segment.SegmentNodeStore

And why do we do an instanceof check at all? Shouldn't DocumentNodeStore be supported as well?

Reply:
This check was already there actually, and this now is needed to check a specific case, where if --verboseRootPath is not present but only --verbose is present - then in case of documentMk we don't need the traversal approach there, since the blobReferenceCollector also gives us the paths to the nodes and hence the check for segmentMK, but if we have --verboseRootPath (finding blobIds under a specific path) , then It needs to do traversal for document mk as well.

Reply:
Ah this is old SegmentStore and new SegmentStore.